docs/site/HasMany-relation.md

```diff
@@ -0,0 +1,161 @@
+repository, the following are required:
+
+- Use [Dependency Injection](Dependency-injection.md) to inject an instance of
+  the target repository in the constructor of your source repository class.
+- Declare a property with the factory function type
+  `HasManyRepositoryFactory<TargetModel, typeof TargetModel.prototype.id>` on
+  the source repository class.
+- call the `_createHasManyRepositoryFactoryFor` function in the constructor of
+  the source repository class with the relation name (decorated relation propert
+  on the source model) and target repository instance and assign it the property
+  mentioned above.
+
+The following code snippet shows how it would look like:
+
+{% include code-caption.html
+content="/src/repositories/customer.repository.ts.ts" %}
+
+```ts
+import {Order, Customer} from '../models';
+import {OrderRepository} from './order.repository.ts';
+import {
+  DefaultCrudRepository,
+  juggler,
+  HasManyRepositoryFactory,
+} from '@loopback/repository';
+
+class CustomerRepository extends DefaultCrudRepository<
+  Customer,
+  typeof Customer.prototype.id
+> {
+  public orders: HasManyRepositoryFactory<Order, typeof Customer.prototype.id>;
+  constructor(
+    @inject('datasources.db') protected db: juggler.DataSource,
+    @repository(OrderRepository) orderRepository: OrderRepository,
+  ) {
+    super(Customer, db);
+    this.orders = this._createHasManyRepositoryFactoryFor(
+      'orders',
+      orderRepository,
+    );
+  }
+}
+```
+
+The following CRUD APIs are now available in the constrained target repository
+factory `orders` for instances of `customerRepository`:
+
+- `create` for creating a target model instance
+- `find` finding target model instance(s)
+- `delete` for deleting target model instance(s)
+- `patch` for patching target model instance(s)
+
+## Using HasMany constrained repository in a controller
+
+The same pattern used for ordinary repositories to expose their CRUD APIs in
+controller methods is employed for `HasMany` repositories. Once the hasMany
+relation has been defined and configured, controller methods can call the
+underlying constrained repository CRUD APIs and expose them as routes once
+decorated with
+[Route decorators](Routes.md#using-route-decorators-with-controller-methods). It
+will require the value of the foreign key and, depending on the request method,
+a partial value for the target model instance as demonstrated below.
+
+{% include code-caption.html
+content="src/controllers/customer-orders.controller.ts" %}
+
+```ts
+import {post, param, requestBody} from '@loopback/rest';
+import {customerRepository} from '../repositories/customer.repository.ts';
+import {Order} from '../models/order.model.ts';
+
+export class CustomerOrdersController {
+  constructor(
+    @repository(CustomerRepository)
+    protected customerRepository: CustomerRepository,
+  ) {}
+
+  @post('/customers/{id}/order')
+  async createCustomerOrders(
+    @param.path.number('id') customerId: number,
```

Comment:
Should we use `typeof Customer.prototype.id` instead of hard-coded `number` here?